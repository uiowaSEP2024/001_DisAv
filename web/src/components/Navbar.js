import React from 'react';
import '../styles/navbar.css';
import { Link } from 'react-router-dom';

const Navbar = ({ isLoggedIn, onLogout }) => {
  return (
    <nav className="navbar" data-testid="navbar">
      <div className="logo">Logo</div>
      <div className="buttons">
        {isLoggedIn ? (
          <>
            <Link to="/dashboard">
              <button>Dashboard</button>
            </Link>
            <button onClick={onLogout}>Log Out</button>
          </>
        ) : (
          <>
            <Link to="/login">
              <button>Log In</button>
            </Link>
            <Link to="/signup">
              <button>Sign Up</button>
            </Link>
          </>
        )}
      </div>
    </nav>
  );
};

export default Navbar;
