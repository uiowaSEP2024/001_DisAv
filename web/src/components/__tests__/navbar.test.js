import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar.js';


test('renders navbar', () => {
  render(
    <Router>
      <Navbar />
    </Router>
  );

  // Check if logo is rendered
  const logoElement = screen.getByText('Logo');
  expect(logoElement).toBeInTheDocument();

  // Check if Log In button is rendered
  const loginButtonElement = screen.getByText('Log In');
  expect(loginButtonElement).toBeInTheDocument();

  // Check if Sign Up button is rendered
  const signupButtonElement = screen.getByText('Sign Up');
  expect(signupButtonElement).toBeInTheDocument();

  // Check if navbar class is present
  const navbarElement = screen.getByRole('navigation');
  expect(navbarElement).toHaveClass('navbar');
});
