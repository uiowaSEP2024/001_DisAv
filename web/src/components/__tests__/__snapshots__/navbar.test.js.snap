// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="logo"
  >
    Logo
  </div>
  <div
    className="buttons"
  >
    <button>
      Log In
    </button>
    <button>
      Sign Up
    </button>
  </div>
</nav>
`;
